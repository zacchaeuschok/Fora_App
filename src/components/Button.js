import React from "react";
import { TouchableOpacity, Text, Image } from "react-native";

import { COLORS, SIZES, FONTS, SHADOWS } from "../constants";

export const CircleButton = ({ imgUrl, handlePress, ...props }) => {
  return (
    <TouchableOpacity
      testID="circle"
      style={{
        width: 40,
        height: 40,
        backgroundColor: COLORS.white,
        position: "absolute",
        borderRadius: SIZES.extraLarge,
        alignItems: "center",
        justifyContent: "center",
        ...SHADOWS.light,
        ...props,
      }}
      onPress={handlePress}
    >
      <Image
        source={imgUrl}
        resizeMode="contain"
        style={{ width: 24, height: 24 }}
      />
    </TouchableOpacity>
  );
};

export const ProfileButton = ({ imgUrl, handlePress, ...props }) => {
  return (
    <TouchableOpacity
      testID="profile"
      style={{
        width: 40,
        height: 40,
        position: "absolute",
        borderRadius: SIZES.extraLarge,
        alignItems: "center",
        justifyContent: "center",
    
        ...props,
      }}
      onPress={handlePress}
    >
      <Image
        source={imgUrl}
        resizeMode="contain"
        style={{ width: 40, height: 40 }}
      />
    </TouchableOpacity>
  );
};

export const ArchiveButton = ({ imgUrl, handlePress, ...props }) => {
  return (
    <TouchableOpacity
      testID="profile"
      style={{
        width: 40,
        height: 40,
        backgroundColor: "transparent",
        position: "absolute",
        borderRadius: SIZES.extraLarge,
        alignItems: "center",
        justifyContent: "center",
        ...props,
      }}
      onPress={handlePress}
    >
      <Image
        source={imgUrl}
        resizeMode="contain"
        style={{ width: 40, height: 40 }}
      />
    </TouchableOpacity>
  );
};

export const RectButton = ({ minWidth, fontSize, handlePress, text, backgroundColor, textColor, ...props }) => {
  return (
    <TouchableOpacity
      testID="rect"
      style={{
        backgroundColor: backgroundColor,
        padding: SIZES.small,
        borderRadius: SIZES.extraLarge,
        minWidth: minWidth,
        ...props,
      }}
      onPress={handlePress}
    >
      <Text
        style={{
          fontFamily: FONTS.semiBold,
          fontSize: fontSize,
          color: textColor,
          textAlign: "center",
        }}
      >
        {text}
      </Text>
    </TouchableOpacity>
  );
};
