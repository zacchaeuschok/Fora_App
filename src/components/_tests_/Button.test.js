import { CircleButton, RectButton, ProfileButton } from "../Button";
import renderer from 'react-test-renderer'
import { fireEvent, render } from '@testing-library/react-native'

jest.useFakeTimers()

it('renders correctly', () => {
    const tree = renderer.create(<CircleButton />)
    const tree2 = renderer.create(<RectButton />)
    const tree3 = renderer.create(<ProfileButton />)
    expect(tree).toMatchSnapshot()
    expect(tree2).toMatchSnapshot()
    expect(tree3).toMatchSnapshot()
  })

  it('on press circle button', () => {
    const mockOnPressFn = jest.fn()
    const { getByTestId } = render(<CircleButton handlePress={mockOnPressFn} />)
    fireEvent.press(getByTestId('circle'))
    expect(mockOnPressFn).toHaveBeenCalled()
  })

  it('on press rect button', () => {
    const mockOnPressFn = jest.fn()
    const { getByTestId } = render(<RectButton handlePress={mockOnPressFn} />)
    fireEvent.press(getByTestId('rect'))
    expect(mockOnPressFn).toHaveBeenCalled()
  })

  it('on press profile button', () => {
    const mockOnPressFn = jest.fn()
    const { getByTestId } = render(<ProfileButton handlePress={mockOnPressFn} />)
    fireEvent.press(getByTestId('profile'))
    expect(mockOnPressFn).toHaveBeenCalled()
  })