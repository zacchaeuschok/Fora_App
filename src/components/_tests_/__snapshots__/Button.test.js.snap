// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#FFF",
      "borderRadius": 24,
      "elevation": 3,
      "height": 40,
      "justifyContent": "center",
      "opacity": 1,
      "position": "absolute",
      "shadowColor": "#74858C",
      "shadowOffset": Object {
        "height": 1,
        "width": 0,
      },
      "shadowOpacity": 0.22,
      "shadowRadius": 2.22,
      "width": 40,
    }
  }
  testID="circle"
>
  <Image
    resizeMode="contain"
    style={
      Object {
        "height": 24,
        "width": 24,
      }
    }
  />
</View>
`;

exports[`renders correctly 2`] = `
<View
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#001F2D",
      "borderRadius": 24,
      "minWidth": undefined,
      "opacity": 1,
      "padding": 12,
    }
  }
  testID="rect"
>
  <Text
    style={
      Object {
        "color": "#FFF",
        "fontFamily": "InterSemiBold",
        "fontSize": undefined,
        "textAlign": "center",
      }
    }
  >
    Place a bid
  </Text>
</View>
`;

exports[`renders correctly 3`] = `
<View
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#FFF",
      "borderRadius": 24,
      "elevation": 3,
      "height": 40,
      "justifyContent": "center",
      "opacity": 1,
      "position": "absolute",
      "shadowColor": "#74858C",
      "shadowOffset": Object {
        "height": 1,
        "width": 0,
      },
      "shadowOpacity": 0.22,
      "shadowRadius": 2.22,
      "width": 40,
    }
  }
  testID="profile"
>
  <Image
    resizeMode="contain"
    style={
      Object {
        "height": 40,
        "width": 40,
      }
    }
  />
</View>
`;
